aiida_pseudo/groups/mixins/cutoffs.py

```diff
@@ -90,72 +89,143 @@ def validate_stringency(self, stringency: str) -> None:
+            specifies the ``cutoff_wfc`` and ``cutoff_rho`` keys, corresponding a float value with the recommended
+            cutoff to be used for the wave functions and charge density, respectively. For example:
+
+                {
+                    "Ag": {
+                        "cutoff_wfc": 50.0,
+                        "cutoff_rho": 200.0
+                    },
+                    ...
+                }
+
+        :param stringency: the stringency corresponding to the provided cutoffs.
         :param unit: string definition of a unit of energy as recognized by the ``UnitRegistry`` of the ``pint`` lib.
-        :raises ValueError: if the cutoffs have an invalid format or the default stringency is invalid.
+            Defaults to electronvolt.
+        :raises ValueError: if the cutoffs have an invalid format or the unit is not a valid energy unit.
         """
         unit = unit or self.DEFAULT_UNIT
 
         self.validate_cutoffs(set(self.elements), cutoffs)
         self.validate_cutoffs_unit(unit)
 
-        if default_stringency is None and len(cutoffs) != 1:
-            raise ValueError('have to explicitly specify a default stringency when specifying multiple cutoff sets.')
+        cutoffs_dict = self._get_cutoffs_dict()
+        cutoffs_dict[stringency] = cutoffs
 
-        default_stringency = default_stringency or list(cutoffs.keys())[0]
+        cutoffs_unit_dict = self._get_cutoffs_unit_dict()
+        cutoffs_unit_dict[stringency] = unit
 
-        self.set_extra(self._key_cutoffs, cutoffs)
-        self.set_extra(self._key_cutoffs_unit, unit)
-        self.set_extra(self._key_default_stringency, default_stringency)
+        self.set_extra(self._key_cutoffs, cutoffs_dict)
+        self.set_extra(self._key_cutoffs_unit, cutoffs_unit_dict)
+        if len(cutoffs_dict) == 1:
+            self.set_default_stringency(stringency)
 
-    def get_cutoffs(self, stringency=None) -> Union[dict, None]:
+    def get_cutoffs(self, stringency=None) -> dict:
         """Return a set of cutoffs for the given stringency.
 
-        :param stringency: optional stringency if different from the default.
-        :return: the cutoffs or ``None`` if no cutoffs whatsoever have been defined for this family.
+        :param stringency: optional stringency for which to retrieve the cutoffs. If not specified, the default
+            stringency of the family is used.
+        :raises ValueError: if no stringency is specified and no default stringency is defined for the family.
+        :raises ValueError: if the requested stringency is not defined for this family.
+        """
+        stringency = stringency or self.get_default_stringency()
+
+        try:
+            return self._get_cutoffs_dict()[stringency]
+        except KeyError as exception:
+            raise ValueError(f'stringency `{stringency}` is not defined for this family.') from exception
+
+    def delete_cutoffs(self, stringency) -> None:
```

Comment:
![oliver](https://user-images.githubusercontent.com/6992332/114423667-b13fc900-9bb7-11eb-9992-98fb540247ab.png)

And maybe we do the same as in `set_cutoffs`, if after the operation is completed there is only one stringency left, that becomes the new default. If you agree, would be good to have a test for this and add in the docstring like for `set_cutoffs`

Reply:
I agree it makes sense, and don't immediately see any issue with this. I wonder if we should print a message when the user removes the `default_stringency` with the `remove_cutoffs` method. Something along the lines of:

* Only one stringency remaining: `The default stringency has been set to: '<LEFTOVER STRINGENCY>'`
* Multiple stringencies remaining: `Default stringency removed but multiple options left. Please set the new default stringency with the ``set_default_stringency()`` method.`

And I will make sure that Oliver's hunger for tests is amply satisfied. 😁

Reply:
Would be fine with me, but what are you suggesting to use for printing the message? I think the only options we have are logging or a `warnings.warn` and probably the latter since we haven't really used or setup logging yet. Printing directly I don't like in normal API code tbh.